src/main/java/co/elastic/support/diagnostics/commands/GenerateLogstashDiagnostics.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ *  or more contributor license agreements. Licensed under the Elastic License
+ *  2.0; you may not use this file except in compliance with the Elastic License
+ *  2.0.
+ */
+package co.elastic.support.diagnostics.commands;
+
+import co.elastic.support.Constants;
+import co.elastic.support.diagnostics.DiagnosticException;
+import co.elastic.support.diagnostics.chain.Command;
+import co.elastic.support.diagnostics.chain.DiagnosticContext;
+import co.elastic.support.rest.RestClient;
+import co.elastic.support.rest.RestResult;
+import co.elastic.support.util.SystemUtils;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import freemarker.template.Configuration;
+import freemarker.template.Template;
+import freemarker.template.TemplateException;
+import freemarker.template.TemplateExceptionHandler;
+import freemarker.template.TemplateModelException;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.IOException;
+import java.io.StringWriter;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+import java.util.Map;
+import java.util.TimeZone;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class GenerateLogstashDiagnostics implements Command {
+
+    private static final Logger logger = LogManager.getLogger(GenerateLogstashDiagnostics.class);
+
+    private static final String DIAGNOSTIC_TEMPLATES_PATH = "logstash-diagnostic-templates/";
+
+    private static final ObjectMapper MAPPER = new ObjectMapper();
+
+    @Override
+    public void execute(DiagnosticContext context) throws DiagnosticException {
+        try {
+            final RestClient client = context.resourceCache.getRestClient(Constants.restInputHost);
+            final JsonData jsonData = new JsonData(getNodeInfo(client), getNodeStats(client));
+            final Configuration configuration = createTemplateConfiguration();
+
+            for (String templateFileName : getAllTemplates()) {
+                final String outputFileName = templateFileName.substring(0, templateFileName.lastIndexOf(".ftlh"));
+                final Path outputFilePath = Paths.get(context.tempDir, "logstash_diagnostic_" + outputFileName);
```

Comment:
Maybe could be extracted in some method that says "target file from template", such as `reportFileNameFromTemplate`

Reply:
I did a small refactoring on the execute method based on your suggestions! Thanks for reviewing :)